pcf/particle/aws/cloudfront/cloudfront.py

```diff
@@ -0,0 +1,154 @@
+# limitations under the License.
+
+from pcf.core.aws_resource import AWSResource
+from pcf.core import State
+from pcf.util import pcf_util
+
+
+class CloudFront(AWSResource):
+    """
+    Particle that maps to AWS Cloud Front
+    """
+
+    flavor = 'cloudfront'
+
+    START_PARAM_FILER = {
+        'CallerReference',
+        'Aliases',
+        'DefaultRootObject',
+        'Origins',
+        'OriginGroups',
+        'DefaultCacheBehavior',
+        'CacheBehaviors',
+        'CustomErrorResponses',
+        'Comment',
+        'Logging',
+        'PriceClass',
+        'Enabled',
+        'ViewerCertificate',
+        'Restrictions',
+        'WebACLId',
+        'HttpVersion',
+        'IsIPV6Enabled'
+    }
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0,
+    }
+
+    UNIQUE_KEYS = ["aws_resource.Comments"]
+
+    def __init__(self, particle_definition):
+        """
+        Args:
+            particle_definition (definition): desired configuration of the particle
+        """
+        super(CloudFront, self).__init__(particle_definition, 'cloudfront')
+        self._id = ""
+
+    def _set_unique_keys(self):
+        """
+        Logic that sets keys from state definition that are used to uniquely identify the distribution
+        """
+        self.unique_keys = CloudFront.UNIQUE_KEYS
+
+    def _start(self):
+        """
+        Creates the distribution according to the particle definition and adds tags if necessary
+        Returns:
+            hosted_zone: boto3 response
+        """
+        start_definition = pcf_util.param_filter(self.desired_state_definition, CloudFront.START_PARAM_FILER)
+        response = self.client.create_distribution_with_tags(
+            DistributionConfigWithTags={
+                "DistributionConfig": start_definition,
+                "Tags": {
+                    "Items": self.desired_state_definition.get("Tags")
+                }
+            }
+        )
+        self._id = response.get("Distribution", {}).get("Id")
+        return response
+
+    def _terminate(self):
+        """
+        Deletes the distribution using its id
+        Returns:
+            boto3 response
+        """
+        return self.client.delete_distribution(Id=self._id)
+
+    def _stop(self):
+        """
+        Calls _terminate
+        """
+        return self._terminate()
+
+    def _update(self):
+        """
+        Not implemented
+        """
+        raise NotImplemented
+
+    def get_status(self):
+        """
+        Gets the current definition of the cloudfront distribution
+        Must list all distributions and search for matching comment field, bc
+        no filtering by name or tagging exists for this resource
+        Returns:
+            current definition of the distribution
+        """
+        response = self.client.list_distributions().get("DistributionList")
+        distribution_list = response.get("Items")
```

Comment:
paginating responses is common across many resources, perhaps look into moving this into a separate utility or class to enable code reuse